src/main/java/user_interface/login_leaderboard/RegisterPanel.java

```diff
@@ -1,14 +1,79 @@
 package user_interface.login_leaderboard;
 
 import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 
 /**
  * Allows the user to create a new account by choosing a username and password.
  * Passes the username, password, and email to the register user controller.
  */
-public class RegisterPanel extends Panel {
+public class RegisterPanel extends Panel implements ActionListener {
+    public JTextField username;
+    public JPasswordField password;
+    public JTextField email;
 
     public RegisterPanel() {
+        this.build();
+        JLabel askUserName = new JLabel("Please enter your username");
+        // check if user already exists, if not continue
+        askUserName.setBounds(270, 30, 450, 70);
+        askUserName.setForeground(this.TEXT_COLOR);
+        askUserName.setFont(this.TEXT_FONT);
```

Comment:
These lines seem to be repeated a few times throughout the class, perhaps you can make a method to assign the settings for askUserName, askEmail, etc. altogether?

Reply:
I can't abstract out the bounds unfortunately since the design of the panel depends on what panel we're working on, but I will get right to moving foreground and font to a new class.